index.js

```diff
@@ -1,85 +1,108 @@
+    botTrap.addEventListener('focus', setFocusToLastFocusableElement);
+
+    function untrap() {
+        if (trappedEl) {
+            topTrap = body.removeChild(topTrap);
+            outerTrapBefore = trappedEl.parentNode.removeChild(outerTrapBefore);
+            innerTrapBefore = trappedEl.removeChild(innerTrapBefore);
+            innerTrapAfter = trappedEl.removeChild(innerTrapAfter);
+            outerTrapAfter = trappedEl.parentNode.removeChild(outerTrapAfter);
+            botTrap = body.removeChild(botTrap);
+
+            trappedEl.classList.remove('keyboard-trap--active');
+
+            // let observers know the keyboard is now trapped
+            var event = document.createEvent('Event');
+            event.initEvent('keyboardUntrap', false, true);
+            trappedEl.dispatchEvent(event);
+
+            trappedEl = null;
+        }
+        return trappedEl;
     }
-    return trappedEl;
-}
 
-function trap(el) {
-    untrap();
+    function trap(el) {
+        untrap();
 
-    trappedEl = el;
+        trappedEl = el;
 
-    var focusableElements = trappedEl.querySelectorAll(util.focusableElementsList);
-    firstFocusableElement = focusableElements[0];
-    lastFocusableElement = focusableElements[focusableElements.length - 1];
+        var focusableElements = trappedEl.querySelectorAll(focusableElementsList);
+        firstFocusableElement = focusableElements[0];
+        lastFocusableElement = focusableElements[focusableElements.length - 1];
 
-    body.insertBefore(topTrap, body.childNodes[0]);
-    trappedEl.parentNode.insertBefore(outerTrapBefore, trappedEl);
-    trappedEl.insertBefore(innerTrapBefore, trappedEl.childNodes[0]);
-    trappedEl.appendChild(innerTrapAfter);
-    trappedEl.parentNode.insertBefore(outerTrapAfter, trappedEl.nextElementSibling);
-    body.appendChild(botTrap);
+        body.insertBefore(topTrap, body.childNodes[0]);
+        trappedEl.parentNode.insertBefore(outerTrapBefore, trappedEl);
+        trappedEl.insertBefore(innerTrapBefore, trappedEl.childNodes[0]);
+        trappedEl.appendChild(innerTrapAfter);
+        trappedEl.parentNode.insertBefore(outerTrapAfter, trappedEl.nextElementSibling);
+        body.appendChild(botTrap);
 
-    // let observers know the keyboard is now trapped
-    var event = document.createEvent('Event');
-    event.initEvent('keyboardTrap', false, true);
-    trappedEl.dispatchEvent(event);
+        // let observers know the keyboard is now trapped
+        var event = document.createEvent('Event');
+        event.initEvent('keyboardTrap', false, true);
+        trappedEl.dispatchEvent(event);
 
-    trappedEl.classList.add('keyboard-trap--active');
+        trappedEl.classList.add('keyboard-trap--active');
 
-    return trappedEl;
-}
+        return trappedEl;
+    }
 
-module.exports = {
-    trap: trap,
-    untrap: untrap
-};
+    return {
+        trap: trap,
+        untrap: untrap,
+        createTrapBoundary: createTrapBoundary,
```

Comment:
Why are we exposing createTrapBoundary in the public API? Is there a need for it?

Reply:
Now that I think of it, I did it for testing, but I don't think the tests use the do they? If not I can remove.